docs/workflow/blocks/form.rst

```diff
@@ -157,4 +157,190 @@ In the example below, a mapping has a default value, which is saved using the co
+                "type": "string",
+                "title": "Password",
+                "default": ""
+            }
+        }
+    },
+    "uiSchema": {
+        "password": {
+            "ui:widget": "password"
+        }
+    }
+  }
+
+
+Nested tasks
+------------
+
+You can insert or "nest" another task within a form through the use of uiSchema.
+First specify your nested task's position in the jsonSchema using the property key of your choice.
+Then you can define the schema's content in the form task with your chosen key, 
+within the enclosing “uiSchema” property.
+
+The below example inserts an array of blocks into the form. Each block displays a simple message.
+
+You can see this example working `here
+<https://app.kendra.io/workflow-builder?data=NobwRALgngDgpmAXGAZgewE4FswBowA2AhgEZwFJgDKAriVgJYR5gBWAzmgHZUDGAFnCxEk4GBjTwMEBnHajIRDAHM4EAPrjJcaVAXR4ldhAwMuyljIgEEyAJpoaAAgDu3AOQQn7OHCcR+BnYnAFonJlcGAgInMic0ADcdUwATFLguWKgnGgY+QWEwAF8S-Fz8oRFEcAglVQ0tKWgFXMQXBhT6yhICNF4Aa3l8Hr7BgGFuFAYLarARgflEUEhYWzAsOXYiVUsmG0oAQScN9i2dotwa1coTs4R8K33kA640AJ1jze2EC6vDZFu3121jWdjUTiIr3eGE+pyBRQAuiULnNegMJlgNlxmMhigigA>`_.
+
+.. code-block:: json
+
+  {
+    "type": "form",
+    "label": "Submit",
+    "jsonSchema": {
+        "properties": {
+            "target_property": {
+                "type": "string",
+                "title": "You won't see this - it will be overridden by uiSchema"
+            }
+        }
+    },
+    "uiSchema": {
+        "target_property": {
+            "ui:widget": "blocks",
+            "blocksConfig": {
+                "blocks": [
+                    {
+                        "type": "message",
+                        "title": "A message"
+                    },
+                    {
+                        "type": "message",
+                        "title": "Another message"
+                    },
+                    {
+                        "type": "message",
+                        "title": "Yet another message"
+                    }
+                ]
+            }
+        }
+    }
+}
+
+Nested flows
+------------
+
+If you want to vary a field according to user input, you can achieve this with a nested flow.
+
+Any flow can be nested in any other flow. The nested flow has access to the main flow's data object, 
+context and state - it can use any data stored here. 
+The nested flow's output is then passed to the main flow's data object, just like the output of a conventional task.
+
+As with the main flow, a nested flow can be edited directly with Kendraio App if opened from the Flow Cloud. 
+Any saved changes will be reflected immediately when the main flow is refreshed.
+
+This example flow allows the user to search and select from a menu based on returned data. 
+The Venue Name field expects a single value and the Lineup field can handle several values.
+
+The nested flow is denoted by the property ``"type": "gosub"``. You can read more about gosubs :doc:`here <gosub>`.
```

Comment:
I am not sure, I understood correctly. Does this mean that a nested flow, is a flow that exists already and is "plugged in" in another flow? So I don't find the flow configuration of the nested flow, in the main flow, right?

Reply:
@gsambrotta That's right - it exists independently of the main flow and is edited separately. I'll have a think about how to make this clearer. If you can think of a better way to explain it, please let me know!

Reply:
Ah OK, all clear. Thank you, also, for making the text more clear.